src/cable/connection.cr

```diff
@@ -167,10 +167,10 @@ module Cable
 
     private def subscribe_to_internal_channel
       server = Cable.server
-      channel = self
+      conn = self
       spawn(name: "Cable::Connection - subscribe_to_internal_channel") do
-        if !channel.connection_rejected?
-          server.backend.open_subscribe_connection(internal_channel)
+        if !conn.closed? || !conn.connection_rejected?
```

Comment:
Should this be &&? This condition would be more clear if made a guard instead of this block. (assuming you did mean &&) it would look like

```crystal
spawn(name: "Cable::Connection - subscxribe_to_internal_channel") do
  break if conn.closed? || conn.connection_rejected?

  server.backend.subscribe(internal_channel)
end
```

Either way, I think the guard is easier to read, expecially in this case, b/c it removes the negations.

Reply:
> A connection can be open and rejected, or it can be closed, so I think `||` is what's needed here

I think what @matthewmcgarvey is talking about is DeMorgan’s Law — [`!(a || b)` is `!a && !b`](https://en.wikipedia.org/wiki/De_Morgan%27s_laws#Negation_of_a_conjunction).

> Can you break from a `spawn` like that? I didn't know you could do that.

`break` is basically `return` for blocks. I _believe_ `next` can also work here (since it only yields once), but `break` is more expressive.